### ISSUE TITLE
[java] TestClassWithoutTestCases - false-positive for test classes extending a class with tests (in nested classes)

### ISSUE THREAD

**Issue by Wolf2323** [pmd/pmd#5011]
**Affects PMD Version:** 7.1.0

**Rule:** [TestClassWithoutTestCases](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#testclasswithouttestcases)

**Description:**
In PMD 6 this was no issue, now in PMD 7 a nested test class is now detected as TestClassWithoutTestCases. I guess previously the class was not detected as test class at all. Now it is detected, and yes it actually has no tests, but they are on the super class.

**Code Sample demonstrating the issue:**

```java
public class SpeedProfilerTests extends TimeProfilerTests {
	@Nested
	class operations extends TimeProfilerTests.operations {
		@Override
		protected TimeProfiler makeProfiler() {
			return new SpeedProfiler("foo");
		}
	}
```

**Expected outcome:**
No violations, as the super class actually has test cases

[INFO] PMD Failure: works.reload.recore.util.operations:22 Rule:TestClassWithoutTestCases Priority:3 The class 'operations' might be a test class, but it contains no test cases..

**Running PMD through:** *Maven*
